### PR TITLE
✨ feat: add guild forum channel support

### PR DESCRIPTION
**Pull request by Helloyunho**
## About

This PR adds support for forum channels.

## Status

- [x] These changes have been tested against Discord API or do not contain API change.
- [ ] This PR includes only documentation changes, no code change.
- [ ] This PR introduces some Breaking changes.

### CODE REVIEW

**Comment by Helloyunho** [comment]
@DjDeveloperr review when

File: src/rest/endpoints.ts [hunk 1]
```diff
@@ -1366,9 +1366,9 @@ The `emoji` must be [URL Encoded](https://en.wikipedia.org/wiki/Percent-encoding
   }
 
   /**
-   * Creates a new public thread from an existing message. Returns a channel on success, and a 400 BAD REQUEST on invalid parameters. Fires a Thread Create Gateway event.
+   * Creates a new thread from an existing message. Returns a channel on success, and a 400 BAD REQUEST on invalid parameters. Fires a Thread Create Gateway event.
    */
-  async startPublicThread(
+  async startPublicThreadFromMessage(
```
Comment (DjDeveloperr) [inline src/rest/endpoints.ts:1371]: Breaking change?

Comment (Helloyunho) [reply src/rest/endpoints.ts:1371]: oh right totally forgot to add back compatibility

**Review by DjDeveloperr — APPROVED** [review]

**Comment by Helloyunho** [comment]
I'll merge this in an hour since there's no any additional comments

**Comment by Helloyunho** [comment]
it definitely passed an hour so im merging this

**Pull request merged**